### PR TITLE
DPDFreduction: documentation for the algorithm

### LINKED ISSUE #16598: DPDFreduction: documentation for the algorithm
Add detailed information of the reduction algorithm regarding physical properties of interest to the user

### PR DESCRIPTION
**Pull request by jmborr**
I updated the documentation (rst file) and grouped the parameters of algorithm DPDFreduction into "Input", "Configuration", and "Output" groups

There is nothing to test, but go over the HTML an QThelp documentation for mistakes or nonsense sentences.

Fixes #16598.

Does not need to be in the release notes.
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by VickieLynch** [comment]
Is this line correct?

ec_S_theta_E: reduced empty can after GroupDetectors algorithm minus empty can ec_S_theta_E.

File: Framework/PythonInterface/plugins/algorithms/DPDFreduction.py [hunk 1]
```diff
@@ -190,20 +209,23 @@ def PyExec(self):
 
         # Convert to energy transfer. Normalize by proton charge.
         # The output workspace is S(detector-id,E)
-        factor = 0.1  # a fine energy bin
+        factor = 0.1  # user a finer energy bin than the one passed (self._ebins[1])
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/DPDFreduction.py]: use not user

**Comment by jmborr** [comment]
@VickieLynch Thanks for spotting that one

**Comment by jmborr** [comment]
@AndreiSavici Thanks Andrei.

**Comment by VickieLynch** [comment]
Rebuilt on analysis machines and documentation and equations look good. :shipit:

**Pull request merged**